pkg/apis/istio/v1alpha1/istiocontrolplane_types.go

```diff
@@ -44,6 +60,115 @@ type IstioControlPlaneList struct {
 	Items           []IstioControlPlane `json:"items"`
 }
 
+// ConditionType represents the type of the condition.  Condition stages are:
+// Initialized, Installed, Reconciled
+type ConditionType string
+
+const (
+	// ConditionTypeInitialized signifies the whether or not the controller has
+	// initialized the CR.
+	ConditionTypeInitialized ConditionType = "Initialized"
+	// ConditionTypeInstalled signifies the whether or not the controller has
+	// installed the resources defined through the CR.
+	ConditionTypeInstalled ConditionType = "Installed"
+	// ConditionTypeReconciled signifies the whether or not the controller has
+	// reconciled the resources defined through the CR.
+	ConditionTypeReconciled ConditionType = "Reconciled"
+)
+
+// ConditionStatus represents the status of the condition
+type ConditionStatus string
+
+const (
+	// ConditionStatusTrue represents completion of the condition, e.g.
+	// Initialized=True signifies that initialization has occurred.
+	ConditionStatusTrue ConditionStatus = "True"
+	// ConditionStatusFalse represents incomplete status of the condition, e.g.
+	// Initialized=False signifies that initialization has not occurred or has
+	// failed.
+	ConditionStatusFalse ConditionStatus = "False"
+	// ConditionStatusUnknown represents unknown completion of the condition, e.g.
+	// Initialized=Unknown signifies that initialization may or may not have been
+	// completed.
+	ConditionStatusUnknown ConditionStatus = "Unknown"
+)
+
+// ConditionReason represents a short message indicating how the condition came
+// to be in its present state.
+type ConditionReason string
+
+const (
+	// ConditionReasonInstallSuccessful ...
+	ConditionReasonInstallSuccessful ConditionReason = "InstallSuccessful"
+	// ConditionReasonInstallError ...
+	ConditionReasonInstallError ConditionReason = "InstallError"
+	// ConditionReasonReconcileSuccessful ...
+	ConditionReasonReconcileSuccessful ConditionReason = "ReconcileSuccessful"
+	// ConditionReasonReconcileError ...
+	ConditionReasonReconcileError ConditionReason = "ReconcileError"
+)
```

Comment:
i was trying to follow the pattern used by other types, e.g. appsv1.Deployment.Status.Conditions.

while a simple status value would be...simple, i think there's value into having conditions.  by having separate conditions for installation and reconciliation, one can determine whether the initial installation failed or if there was a failure processing later updates.  i think this would be useful information to have, especially because processing updates to the CR could be problematic.  do you mind if we leave it for now?

Reply:
we can address this in a future pr